File: tool/src/org/antlr/v4/automata/EpsilonTransition.java
package org.antlr.v4.automata;

public class EpsilonTransition extends Transition {
	public EpsilonTransition(NFAState target) { super(target); }

	public boolean isEpsilon() { return true; }	

	public int compareTo(Object o) {
		return 0;
	}
}

File: runtime/JavaScript/src/antlr4/transition/EpsilonTransition.js
import Transition from "./Transition.js";

export default class EpsilonTransition extends Transition {
    constructor(target, outermostPrecedenceReturn) {
        super(target);
        this.serializationType = Transition.EPSILON;
        this.isEpsilon = true;
        this.outermostPrecedenceReturn = outermostPrecedenceReturn;
    }

    matches(symbol, minVocabSymbol, maxVocabSymbol) {
        return false;
    }

    toString() {
        return "epsilon";
    }
}
